bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -220,12 +212,22 @@ protected void cacheStateType(String stateName, int type) {
                 default:
                     return null;
             }
-        } catch (NumberFormatException ex) {
+        } catch (IllegalArgumentException ex) {
             logger.debug("{} Error while parsing Tahoma state! Value: {} type: {}", url, state.getValue(), type, ex);
         }
         return null;
     }
 
+    private int normalizePercent(Double valPct) {
+        int value = valPct.intValue();
+        if (value < 0) {
+            value = 0;
+        } else if (value > 100) {
+            value = 0;
```

Comment:
Why not cap the value at 100 instead of treating it as a 0?
Another option would be to return an UnDefType.

Reply:
you are right, it should be 100.

Reply:
Fixed